skyportal/handlers/api/gcn.py

```diff
@@ -167,11 +167,28 @@ def post_gcnevent_from_xml(payload, user_id, session):
     try:
         ra, dec, error = (float(val) for val in skymap["localization_name"].split("_"))
         if error < SOURCE_RADIUS_THRESHOLD:
-            source = {
-                'id': Time(event.dateobs).isot.replace(":", "-"),
-                'ra': ra,
-                'dec': dec,
-            }
+            if any([True if 'GRB' in tag.text.upper() else False for tag in tags]):
+                dateobs_txt = Time(dateobs).isot
+                source_name = f"GRB{dateobs_txt[2:4]}{dateobs_txt[5:7]}{dateobs_txt[8:10]}.{dateobs_txt[11:13]}{dateobs_txt[14:16]}{dateobs_txt[17:19]}"
+                source = {
+                    'id': source_name,
+                    'ra': ra,
+                    'dec': dec,
+                }
+            elif any([True if 'GW' in tag.text.upper() else False for tag in tags]):
```

Comment:
Did you see if there was a GW specific name in the gcn event?

Reply:
Not yet! Here's what I'll do: If the VOEvent notice has a **Name** tag, I'll use that name. Otherwise, it will be a name like this one. What do you think?

Reply:
I can live with that. I expect we will evolve this as the IGWN stuff settles.